packages/office-ui-fabric-react/src/components/HoverCard/HoverCard.tsx

```diff
@@ -0,0 +1,161 @@
+/* tslint:disable:no-unused-variable */
+import * as React from 'react';
+/* tslint:enable:no-unused-variable */
+import {
+  BaseComponent,
+  css,
+  getNativeProps,
+  divProperties,
+  customizable,
+  autobind,
+  getRTL
+} from '../../Utilities';
+import { IHoverCardProps, IHoverCardStyles } from './HoverCard.Props';
+import { Callout, ICallout } from '../../Callout';
+import { DirectionalHint } from '../../common/DirectionalHint';
+import { AnimationClassNames, mergeStyles } from '../../Styling';
+
+import { getStyles } from './HoverCard.styles';
+
+export enum HoverCardMode {
+  compact,
+  expanded
+}
+
+export interface IHoverCardState {
+  mode: HoverCardMode;
+  needsScroll: boolean;
+}
+
+@customizable(['theme'])
+export class HoverCard extends BaseComponent<IHoverCardProps, IHoverCardState> {
+  public static defaultProps = {
+    expandedCardOpenDelay: 1000,
+    openExpanded: false,
+    compactCardHeight: 156,
+    expandedCardHeight: 384
+  };
+
+  private _styles: IHoverCardStyles;
+  private _callout: ICallout;
+  private _expandedElem: HTMLDivElement;
+
+  constructor(props: IHoverCardProps) {
+    super(props);
+
+    this.state = {
+      mode: props.openExpanded ? HoverCardMode.expanded : HoverCardMode.compact,
+      needsScroll: false
+    };
+  }
+
+  public componentWillMount() {
+    this._async.setTimeout(() => {
+      this.setState({
+        mode: HoverCardMode.expanded
+      });
+    }, this.props.expandedCardOpenDelay);
+  }
+
+  public componentDidMount() {
+    if (this._expandedElem && this._expandedElem.scrollHeight >= this.props.expandedCardHeight) {
+      this.setState({
+        needsScroll: true
+      });
+    }
+  }
+
+  public componentWillUpdate(newProps: IHoverCardProps, newState: IHoverCardState) {
+    if (newProps.openExpanded !== this.props.openExpanded) {
+      this.setState({
+        mode: newProps.openExpanded ? HoverCardMode.expanded : HoverCardMode.compact
+      });
+    }
+  }
+
+  public componentWillUnmount() {
+    this._async.dispose();
+  }
+
+  public render() {
+    const {
+      targetElement,
+      id,
+      theme,
+      styles: customStyles,
+      onRenderCompactContent,
+      onRenderExpandedContent,
+      compactCardHeight,
+      expandedCardHeight
+    } = this.props;
+    this._styles = getStyles(theme, customStyles);
+
+    return (
+      <Callout
+        componentRef={ c => this._callout = c }
+        className={ css(
+          AnimationClassNames.scaleUpIn100,
+          this._styles.root
+        ) }
+        targetElement={ targetElement }
+        { ...getNativeProps(this.props, divProperties) }
+        isBeakVisible={ false }
+        directionalHint={ getRTL() ? DirectionalHint.bottomRightEdge : DirectionalHint.bottomLeftEdge }
+        directionalHintFixed={ true }
+        finalHeight={ compactCardHeight + expandedCardHeight }
+        minPagePadding={ 24 }
+      >
+        <div
+          { ...{ onFocusCapture: this.props.onEnter } }
```

Comment:
{ ...{ onFocusCapture: this.props.onEnter } } === { onFocusCapture: this.props.onEnter }

Do we need a spread here?